### DISCUSSION TITLE
Compatibility with Phone 2(a)

### DISCUSSION THREAD

**Discussion by Eliot-BPJ** [SebiAi/custom-nothing-glyph-tools#28] [General]
Hello there,

I've just discovered that we can have custom ringtones and tried to download some from internet.
I can't import all of them cause of my Phone 2(a), and I understand that it can't because their glyph interfaces are differents.

But I've still managed to import ringtones with custom sounds, so I was asking myself what were the differences, and if they were made with this tool or another one, as it's written in the readme that it's not compatible with the 2(a)

Thx in advance

**Comment by SebiAi** [comment]
To be begin with, I think I need to differentiate two things: The metadata in the audio and the tools themselves.
The format aka. how the metadata (the individual light values as csv values) was a given by the *Glyph Composer* app - the tools/scripts just facilitate a better way to create, write and read that data.

With that out of the way, I can give you a more in depth answer:
The tools do not currently support the "create and write" processes for the Phone (2a) which I mentioned above (reading is an exception because it is the same for all models - for now).
It seems like Nothing implemented some kind of backwards compatible way of playing back a composition made for Phone (2) on the Phone (2a) which I did not investigate yet properly.

Some differences I can tell you right now (Phone (2a)):
* We have 26 addressable LED's
* Index 0-23 are in the top left Glyph where index 0 is the LED at the bottom left of the Glyph
* Index 24 is the middle right Glyph
* Index 25 is the bottom left Glyph

On another note:
You can even force the playback of a Phone (2) composition on a Phone (1) by placing the composition in the `Ringtones/Compositions` folder but most of the time it just does nothing and the other times it just completely spasms out.

Now a question for you: How did you manage to import these compositions you talked about? Did it import fine via *Glyph Composer*, did you use the folder trick mentioned above or did you do it some other kind of way?

I hope this answers your question.

**Reply by Eliot-BPJ** [reply]
Thank you for your answer it's way more clear now 😀 

I've downloaded some on this website: https://miguel-antunes.outsystemscloud.com/GlyphComposerSoundSphere/Ringtones
The ones using the tag 'other' are not working at all and i can't import them, but for example I can use the 'Minecraft Theme' made by NTND, using piano notes not available in original sample packs

And yes I'm using Glyph Composer import option
